core/src/main/java/org/elasticsearch/index/analysis/SynonymTokenFilterFactory.java

```diff
@@ -23,82 +23,151 @@
 import org.apache.lucene.analysis.core.LowerCaseFilter;
-import org.apache.lucene.analysis.core.WhitespaceTokenizer;
 import org.apache.lucene.analysis.synonym.SolrSynonymParser;
 import org.apache.lucene.analysis.synonym.SynonymFilter;
 import org.apache.lucene.analysis.synonym.SynonymMap;
 import org.apache.lucene.analysis.synonym.WordnetSynonymParser;
+import org.apache.lucene.util.IOUtils;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.assistedinject.Assisted;
 import org.elasticsearch.common.io.FastStringReader;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.indices.analysis.AnalysisModule;
 
+import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.Reader;
 import java.util.List;
 import java.util.Map;
 
 public class SynonymTokenFilterFactory extends AbstractTokenFilterFactory {
 
-    private final SynonymMap synonymMap;
     private final boolean ignoreCase;
 
-    public SynonymTokenFilterFactory(IndexSettings indexSettings, Environment env, Map<String, TokenizerFactory> tokenizerFactories,
-                                      String name, Settings settings) throws IOException {
+    public SynonymTokenFilterFactory(IndexSettings indexSettings, Environment env, AnalysisRegistry analysisRegistry,
+                                     Map<String, TokenizerFactory> tokenizerFactories, String name, Settings settings) throws IOException {
         super(indexSettings, name, settings);
 
-        Reader rulesReader = null;
         if (settings.getAsArray("synonyms", null) != null) {
-            List<String> rules = Analysis.getWordList(env, settings, "synonyms");
+            List<String> rulesList = Analysis.getWordList(env, settings, "synonyms");
             StringBuilder sb = new StringBuilder();
-            for (String line : rules) {
+            for (String line : rulesList) {
                 sb.append(line).append(System.lineSeparator());
             }
-            rulesReader = new FastStringReader(sb.toString());
+            rules = sb.toString();
         } else if (settings.get("synonyms_path") != null) {
-            rulesReader = Analysis.getReaderFromFile(env, settings, "synonyms_path");
+            Reader rulesReader = Analysis.getReaderFromFile(env, settings, "synonyms_path");
+            BufferedReader br = new BufferedReader(rulesReader);
+            StringBuilder sb = new StringBuilder();
+            String tmp;
+            try {
+                while ((tmp = br.readLine()) != null) {
+                    sb.append(tmp).append(System.getProperty("line.separator"));
+                }
+                rules = sb.toString();
+            } catch (IOException ioe) {
+                throw new IllegalArgumentException("failed to load synonyms", ioe);
+            } finally {
+                IOUtils.closeWhileHandlingException(br);
```

Comment:
I don't think we should swallow the exceptions here, instead, this could be a try-with-resources block like:

```java
try (BufferedReader br = new BufferedReader(rulesReader)) {
    .. read the string ..
}
```